pkg/api/plugin_checks.go

```diff
@@ -28,7 +28,13 @@ func checkAppEnabled(pluginStore pluginstore.Store, pluginSettings pluginsetting
 		})
 		if err != nil {
 			if errors.Is(err, pluginsettings.ErrPluginSettingNotFound) {
-				c.JsonApiErr(http.StatusNotFound, "Plugin not found", nil)
+				// If the plugin is auto enabled, we don't want to return an error because auto enabling allows us
+				// to enable plugins that are not explicitly configured.
+				if p.AutoEnabled {
+					return
+				}
+
+				c.JsonApiErr(http.StatusNotFound, "Plugin setting not found", nil)
```

Comment:
Note: I also updated the wording here as it was confusing initially when doing an RCA. Not sure if it was an oversight originally or on purpose

Reply:
Looks like a copy-paste error from line 18 so this is fine